src/ui/hooks/useEditor.ts

```diff
@@ -72,6 +79,35 @@ export function useEditor(): UseEditor {
     [mainThread]
   );
 
+  const goToRef = useCallback(
+    (resourceId: number) => {
+      const resource = getLocalResource<LocalResourceInstance<ResourceDefinition, IMainThreadContext>>(
+        mainThread,
+        resourceId
+      );
+
+      if (!resource) {
+        return;
+      }
+
+      const resourceType = resource.constructor;
+
+      setSelectedEntity(mainThread, resourceId);
+      setState((prev) => ({
+        ...prev,
+        hierarchyTab: HierarchyTab.Resources,
+        resourceType,
+        resources: buildResourceList(mainThread, resourceType),
+      }));
+
+      if (treeViewRef) {
+        // TODO: Figure out why this doesn't seem to be scrolling to the node properly
```

Comment:
Note that we still need to look into this

Reply:
Yeah, will look into this when adding back button in property panel.